### PR TITLE
Add e2edev tests support for ppc64le arch

### LINKED ISSUE #2366: Add ppc64le support for e2edev tests
Extra effort which needs to make sure agent's changes are obviously consistent with ppc64le arch runtime for agent and its workloads - it may be less necessary if e2edev tests are covering this arch support too.

### PR DESCRIPTION
**Pull request by jacob-rogers**
Signed-off-by: Dmitry Gorbachev <dmitry.gorbachev@ru.ibm.com>

Co-authored-by: Yuri Pravosud <yuri.pravosud2@ibm.com>

Resolves #2366

### CODE REVIEW

**Comment by dabooz** [comment]
@jacob-rogers Please open an issue for this PR and attach this PR to that issue so that we can review it.

**Comment by jacob-rogers** [comment]
@dabooz Issue is opened and linked https://github.com/open-horizon/anax/issues/2366

File: test/gov/run_kube.sh [hunk 1]
```diff
@@ -99,6 +99,14 @@ fi
 EX_IP=${EX_IP} CSS_IP=${CSS_IP} envsubst < "${AGBOT_TEMPFS}/etc/agent-in-kube/horizon.env" > "${AGBOT_TEMPFS}/etc/agent-in-kube/horizon"
 if [ $? -ne 0 ]; then echo "Failure configuring agent env var file"; exit 1; fi
 
+ARCH=${ARCH} envsubst < "${AGBOT_TEMPFS}/etc/agent-in-kube/deployment.yaml.tmpl" > "${AGBOT_TEMPFS}/etc/agent-in-kube/deployment.yaml"
+if [ $? -ne 0 ]; then echo "Failure configuring k8s agent deployment template file"; exit 1; fi
+
+ARCH=${ARCH} envsubst < "$PWD/gov/input_files/k8s_deploy/node_ui.json.tmpl" > "$PWD/gov/input_files/k8s_deploy/node_ui.json"
+if [ $? -ne 0 ]; then echo "Failure configuring k8s agent node userinput file"; exit 1; fi
+
+$PWD/gov/input_files/k8s_deploy/node_ui.json
```
Comment (linggao) [inline test/gov/run_kube.sh:108]: What are you trying to do here?

File: test/gov/agbot_apitest.sh [hunk 2]
```diff
@@ -5,12 +5,22 @@ USERDEV_ADMIN_AUTH="userdev/userdevadmin:userdevadminpw"
...
+  echo -e "\n${PREFIX} Envvar AGBOT_SAPI_URL is empty. Skip test\n"
+  exit 0
+fi
+
 COMP_RESULT=""
 
+# setup ARCH in all policies, patterns & service definition files for deploycheck test
+for in_file in "business_pol_location.json" "service_policy.json" "node_ui.json" "pattern_sloc.json" "service_location.json" "service_locgps.json"
+do
+  sed -i -e "s#__ARCH__#${ARCH}#g" in_file
```
Comment (linggao) [inline test/gov/agbot_apitest.sh:21]: Got the following errors here:
sed: can't read in_file: No such file or directory
sed: can't read in_file: No such file or directory
sed: can't read in_file: No such file or directory
sed: can't read in_file: No such file or directory
sed: can't read in_file: No such file or directory
sed: can't read in_file: No such file or directory

File: test/gov/run_kube.sh [hunk 3]
```diff
@@ -245,7 +253,6 @@ fi
 $cprefix microk8s.kubectl cp $PWD/gov/input_files/k8s_deploy/node.policy.json ${NAME_SPACE}/${POD}:/home/agentuser/.
 $cprefix microk8s.kubectl cp $PWD/gov/input_files/k8s_deploy/node_ui.json ${NAME_SPACE}/${POD}:/home/agentuser/.
 
-#$cprefix microk8s.kubectl exec ${POD} -it -n ${NAME_SPACE} -- /usr/bin/hzn register -f /home/agentuser/node_ui.json --policy /home/agentuser/node.policy.json -u root/root:${EXCH_ROOTPW}
 $cprefix microk8s.kubectl exec ${POD} -it -n ${NAME_SPACE} -- /usr/bin/hzn register -f /home/agentuser/node_ui.json -p e2edev@somecomp.com/sk8s -u root/root:${EXCH_ROOTPW}
```
Comment (linggao) [inline test/gov/run_kube.sh:256]: Pass in ARCH to node_ui.json like this: 
````
$cprefix microk8s.kubectl exec ${POD} -it -n ${NAME_SPACE} -- env ARCH=${ARCH} /usr/bin/hzn register -f /home/agentuser/node_ui.json -p e2edev@somecomp.com/sk8s -u root/root:${EXCH_ROOTPW}
````

File: test/gov/service_upgrading_downgrading_test.sh [hunk 4]
```diff
@@ -130,7 +132,7 @@ cat <<EOF >$KEY_TEST_DIR/svc_cpu.json
   "deployment":{
     "services":{
       "cpu":{
-        "image":"openhorizon/amd64_cpu:1.2.2",
+        "image":"${CPU_IMAGE_NAME}:${CPU_IMAGE_TAG}"
```
Comment (linggao) [inline test/gov/service_upgrading_downgrading_test.sh:135]: Need a comma at the end.

**Comment by linggao** [comment]
Please add the following code in gov-combined.sh at line 292 before `echo "TEST_PATTERNS=${TEST_PATTERNS}"`
````
for in_file in "/root/input_files/compcheck/*.json"
do
  sed -i -e "s#__ARCH__#${ARCH}#g" $in_file
done

for in_file in "/root/objects/*.json"
do
  sed -i -e "s#__ARCH__#${ARCH}#g" $in_file
done
````

File: test/gov/resource_package.sh [hunk 5]
```diff
@@ -37,6 +37,11 @@ for dir in */; do
 			exit -1
 		fi
 	else
+		# Create policy files from templates by passing current ARCH to them
+		for in_file in "/root/objects/basicres.policy" "/root/objects/multires.policy"
+		do
+			sed -i -e "s#__ARCH__#${ARCH}#g" in_file
+		done
```
Comment (linggao) [inline test/gov/resource_package.sh:44]: Please remove above code and see my last comment.

File: test/gov/agbot_apitest.sh [hunk 6]
```diff
@@ -5,12 +5,22 @@ USERDEV_ADMIN_AUTH="userdev/userdevadmin:userdevadminpw"
...
+fi
+
 COMP_RESULT=""
 
+# setup ARCH in all policies, patterns & service definition files for deploycheck test
+for in_file in "business_pol_location.json" "service_policy.json" "node_ui.json" "pattern_sloc.json" "service_location.json" "service_locgps.json"
+do
+  sed -i -e "s#__ARCH__#${ARCH}#g" /root/input_files/compcheck/$in_file
+done
+
```
Comment (linggao) [inline test/gov/agbot_apitest.sh:23]: Please remove line 18-23 and see my last comment.

**Review by linggao — CHANGES_REQUESTED** [review]
per comments

**Comment by linggao** [comment]
@jacob-rogers @ypravo could you please push again? The travis tests failed and I cannot restart the tests.

**Comment by jacob-rogers** [comment]
> @jacob-rogers @ypravo could you please push again? The travis tests failed and I cannot restart the tests.

@linggao Done. And changes from your comments were also included. But please review all changes again if you can - few additional files needed to be fixed.

**Review by linggao — APPROVED** [review]

**Pull request merged**